ra-optimisation/linear-range-action-rao/src/main/java/com/farao_community/farao/linear_range_action_rao/LinearRangeActionRao.java

```diff
@@ -35,7 +51,60 @@ public String getVersion() {
 
     @Override
     public CompletableFuture<RaoComputationResult> run(Network network, Crac crac, String variantId, ComputationManager computationManager, RaoParameters parameters) {
-        // TODO : implement LinearRangeActionRao
         return null;
     }
+
+    public CompletableFuture<RaoComputationResult> run(Network network, Crac crac, String variantId,
+                                                       ComputationManager computationManager, RaoParameters parameters,
+                                                       SensitivityComputationFactory sensitivityComputationFactory) {
+        SensitivityComputationService.init(sensitivityComputationFactory, computationManager);
+        SensitivitySecurityAnalysisResult sensiSaResults = SensitivitySecurityAnalysisService.runSensitivity(network, crac, computationManager);
+
+        SensitivityComputationResults preSensi = sensiSaResults.getPrecontingencyResult();
+        PreContingencyResult preRao = new PreContingencyResult();
+        // 1. do for pre
+        // get cnec from crac
+        crac.getCnecs().forEach(cnec -> {
```

Comment:
After crac refactoring you will be able to select the preventive cnecs here

Reply:
ok